model/trigger/version.go

```diff
@@ -129,23 +138,76 @@ func (t *versionTriggers) versionOutcome(sub *event.Subscription) (*notification
 		return nil, nil
 	}
 
-	return t.generate(sub)
+	return t.generate(sub, "")
 }
 
 func (t *versionTriggers) versionFailure(sub *event.Subscription) (*notification.Notification, error) {
 	if t.data.Status != evergreen.VersionFailed {
 		return nil, nil
 	}
 
-	return t.generate(sub)
+	return t.generate(sub, "")
 }
 
 func (t *versionTriggers) versionSuccess(sub *event.Subscription) (*notification.Notification, error) {
 	if t.data.Status != evergreen.VersionSucceeded {
 		return nil, nil
 	}
 
-	return t.generate(sub)
+	return t.generate(sub, "")
+}
+
+func (t *versionTriggers) versionExceedsDuration(sub *event.Subscription) (*notification.Notification, error) {
+	if t.data.Status != evergreen.VersionSucceeded && t.data.Status != evergreen.VersionFailed {
+		return nil, nil
+	}
+	thresholdString, ok := sub.TriggerData[event.VersionDurationKey]
+	if !ok {
+		return nil, fmt.Errorf("subscription %s has no build time threshold", sub.ID)
+	}
+	threshold, err := strconv.Atoi(thresholdString)
+	if err != nil {
+		return nil, fmt.Errorf("subscription %s has an invalid time threshold", sub.ID)
+	}
+
+	maxDuration := time.Duration(threshold) * time.Second
+	if !t.version.StartTime.Add(maxDuration).Before(t.version.FinishTime) {
+		return nil, nil
+	}
+	return t.generate(sub, fmt.Sprintf("exceeded %d seconds", threshold))
+}
+
+func (t *versionTriggers) versionRuntimeChange(sub *event.Subscription) (*notification.Notification, error) {
+	if t.data.Status != evergreen.VersionSucceeded && t.data.Status != evergreen.VersionFailed {
+		return nil, nil
+	}
+	percentString, ok := sub.TriggerData[event.VersionPercentChangeKey]
+	if !ok {
+		return nil, fmt.Errorf("subscription %s has no percentage increase", sub.ID)
+	}
+	percent, err := strconv.ParseFloat(percentString, 64)
+	if err != nil {
+		return nil, fmt.Errorf("subscription %s has an invalid percentage", sub.ID)
+	}
+
+	lastGreen, err := t.version.LastSuccessful()
+	if err != nil {
+		return nil, errors.Wrap(err, "error retrieving last green build")
+	}
+	if lastGreen == nil {
+		return nil, nil
+	}
+	thisVersionDuration := float64(t.version.FinishTime.Sub(t.version.StartTime))
+	prevVersionDuration := float64(lastGreen.FinishTime.Sub(lastGreen.StartTime))
+	ratio := thisVersionDuration / prevVersionDuration
+	if !util.IsFiniteNumericFloat(ratio) {
+		return nil, nil
+	}
+	percentChange := math.Abs(100*ratio - 100)
+	if percentChange < percent {
+		return nil, nil
+	}
+	return t.generate(sub, fmt.Sprintf("changed in runtime by %f%% (over threshold of %f%%)", percentChange, percent))
```

Comment:
Round percent in string format, maybe `%.2f%%`?

Reply:
Done, and I'll round the other triggers in EVG-3535